### PR TITLE
Reject promise if amp-share-tracking is disabled

### PR DESCRIPTION
**Pull request by muxin**

### CODE REVIEW

File: extensions/amp-share-tracking/0.1/amp-share-tracking.js [hunk 1]
```diff
@@ -43,6 +43,11 @@ export class AmpShareTracking extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
+    if (!this.isExperimentOn_()) {
+      getService(this.win, 'share-tracking',
+          () => Promise.reject(user().createError(TAG + ' disabled')));
+    }
+
     user().assert(this.isExperimentOn_(), `${TAG} experiment is disabled`);
```
Comment (lannka) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: hmm... if we have to do this, i prefer to assert in the if block.

**Comment by lannka** [comment]
LGTM with one comment

File: extensions/amp-share-tracking/0.1/amp-share-tracking.js [hunk 2]
```diff
@@ -43,6 +43,11 @@ export class AmpShareTracking extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
+    if (!this.isExperimentOn_()) {
+      getService(this.win, 'share-tracking',
+          () => Promise.reject(user().createError(TAG + ' disabled')));
```
Comment (jridgewell) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: Resolve for the same reasoning as https://github.com/ampproject/amphtml/pull/4498#discussion_r74815193.

Comment (lannka) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: This is not a rendering block extension. So it should be fine?

Comment (jridgewell) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: Oh, my mistake. This one is just used for url replacements?

Comment (muxin) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: Yeah, just for url replacement

Comment (jridgewell) [inline extensions/amp-share-tracking/0.1/amp-share-tracking.js]: As long as URL experiments can handle a rejection without, fine with me.

**Pull request merged**